src/Uno.UI/UI/Xaml/Controls/Canvas/Canvas.Layout.cs

```diff
@@ -13,20 +13,8 @@ public partial class Canvas : ICustomClippingElement
 	{
 		protected override Size MeasureOverride(Size availableSize)
 		{
-			double maxWidth = 0, maxHeight = 0;
-
-			foreach (var child in Children.Where(c => c is DependencyObject))
-			{
-				var childX = GetLeft(child as DependencyObject);
-				var childY = GetTop(child as DependencyObject);
-
-				var measuredSize = MeasureElement(child, new Size(double.PositiveInfinity, double.PositiveInfinity));
-
-				maxHeight = Math.Max(maxHeight, measuredSize.Height + childY);
-				maxWidth = Math.Max(maxWidth, measuredSize.Width + childX);
-			}
-
-			return new Size(maxWidth, maxHeight);
+			// A canvas does not have dimensions and will always return zero even with a chidren collection.
+			return new Size(0, 0);
```

Comment:
If the children aren't measured here, then their `DesiredSize` will never be set. You should do (and add) some tests to see whether the children get measured (even if the `Canvas` doesn't use the measurement).

Reply:
True. I request now a MeasuredElement for the child otherwise they will not get their DesiredSize set